utils/db.py

```diff
@@ -92,12 +93,14 @@ class Db(Mapping):
         tables, like the mapping interface or :py:meth:`values`.
 
     """
-    def __init__(self, hostname=None, credentials=None):
+    def __init__(self, hostname=None, credentials=None, db_port=None):
         self._table_cache = {}
-        if hostname is None:
+        if hostname is None and db_port is None:
```

Comment:
Well I guess the only case where it isn't is in the case of an external DB, but even then the data for it should be retrieved from an appliance right?